compiler/src/dmd/iasmgcc.d

```diff
@@ -76,19 +84,19 @@ public Statement gccAsmSemantic(GccAsmStatement s, Scope* sc)
     {
         foreach (i; 0 .. s.args.length)
         {
-            Expression e = (*s.args)[i];
-            e = e.expressionSemantic(sc);
+            Expression ec = (*s.constraints)[i];
+            ec = ec.expressionSemantic(sc);
+            assert(ec.op == EXP.string_ && (cast(StringExp) ec).sz == 1);
+            (*s.constraints)[i] = ec;
+
+            Expression earg = (*s.args)[i];
+            earg = earg.expressionSemantic(sc);
             // Check argument is a valid lvalue/rvalue.
             if (i < s.outputargs)
-                e = e.modifiableLvalue(sc);
-            else if (e.checkValue())
-                e = ErrorExp.get();
-            (*s.args)[i] = e;
-
-            e = (*s.constraints)[i];
-            e = e.expressionSemantic(sc);
-            assert(e.op == EXP.string_ && (cast(StringExp) e).sz == 1);
-            (*s.constraints)[i] = e;
+                earg = earg.modifiableLvalue(sc);
+            else if (earg.checkValue())
+                earg = ErrorExp.get();
+            (*s.args)[i] = earg;
```

Comment:
Operands are parsed in the following order:
```
[Name] Constraint (Arg)
```
So then do semantic analysis of each part from left to right, not in reverse.